chessticulate_api/routers/v1/main.py

```diff
@@ -39,9 +40,16 @@ async def login(payload: schemas.LoginRequest) -> schemas.LoginResponse:
 
 @router.post("/signup")
 async def signup(payload: schemas.CreateUserRequest) -> schemas.CreateUserResponse:
-    return dict(
-        await crud.create_user(payload.name, payload.email, SecretStr(payload.password))
+    # this if statement is not properly handling bad credentials
+    if not (
+        user := await crud.create_user(payload.name, payload.email, payload.password)
+    ):
+        raise HTTPException(status_code=401, detail="invalid credentials")
+
+    response_data = schemas.CreateUserResponse(
+        name=user.name, email=user.email, password=user.password
     )
+    return dict(response_data)
```

Comment:
I would change this to status_code=400 and detail="account with username or email already exists". A 401 status code is usually for authentication, whereas this is about creating a new piece of data (a user).